src/main/java/com/hazelcast/security/CompactCredentialsLoginModule.java

```diff
@@ -0,0 +1,45 @@
+package com.hazelcast.security;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+import java.io.IOException;
+
+public class CompactCredentialsLoginModule extends ClusterLoginModule {
+    private String name;
+
+    @Override
+    protected boolean onLogin() throws LoginException {
+        CredentialsCallback cb = new CredentialsCallback();
+        TokenDeserializerCallback tdcb = new TokenDeserializerCallback();
+        try {
+            callbackHandler.handle(new Callback[]{cb, tdcb});
+        } catch (IOException | UnsupportedCallbackException e) {
+            throw new LoginException("Problem getting credentials");
+        }
+        Credentials credentials = cb.getCredentials();
+        if (credentials instanceof TokenCredentials) {
+            TokenCredentials tokenCreds = (TokenCredentials) credentials;
+            credentials = (Credentials) tdcb.getTokenDeserializer().deserialize(tokenCreds);
+        }
+        if (!(credentials instanceof CompactCredentials)) {
+            throw new FailedLoginException();
```

Comment:
Can you also put detailed description of the failure? possibly printing credentials if not null.

Reply:
OK, let me do that. I just copy-pasted the login module from the https://github.com/hazelcast/hazelcast-enterprise/blob/master/hazelcast-enterprise/src/test/java/com/hazelcast/client/security/ClientCustomAuthenticationTest.java#L330